core/src/test/java/com/netease/arctic/io/TestIcebergCombinedReader.java

```diff
@@ -150,6 +167,16 @@ public void initDataAndReader() throws IOException {
             new DeleteFile[] {},
             new DeleteFile[] {},
             getArcticTable());
+    filterEqDeleteScanTask =
+        new RewriteFilesInput(
+            new DataFile[] {MixedDataTestHelpers.wrapIcebergDataFile(dataFile, 1L)},
+            new DataFile[] {},
```

Comment:
Unit tests need to cover scenarios with rePosDeletedDataFiles. BTW, This is a important improvement, It would be best to add this case within com.netease.arctic.server.optimizing.flow.

Reply:
Yes, I will add Unit tests to cover scenarios with rePosDeletedDataFiles.